coderd/database/migrations/000163_external_auth_extra.up.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE external_auth_links ADD COLUMN "oauth_extra" jsonb;
```

Comment:
Should we just make this `NOT NULL DEFAULT '{}'`?

Dealing with `null` might be more annoying then just returning an empty map?

Reply:
I think it's good to know the user intentionally didn't want to store any fields.

Reply:
The oauth config is always required to use the link, so the `ExtraKeys` should always be accessible to the user when they are looking at the link.

The other thing that it might be beneficial for is if the config is updated to include keys. The old links will be set to `null` and new links will have a json payload. 

So on the read side, we will need to switch on the `Valid` always if `Keys` are set. This could be an argument for keeping it `nullable` though, so I am good either way.

Reply:
It's mildly easier to migrate to have a default of `{}` from null then it is to go the other way, so let's keep it as-is for now and if it's weird to interface with I'll happily change.